### ISSUE TITLE
自从0.6.1 bert不再支持多gpu

### ISSUE THREAD

**Issue by tiandiweizun** [bojone/bert4keras#175]
提问时请尽可能提供如下信息：

### 基本信息
- 你使用的**操作系统**: linux
- 你使用的**Python**版本: 3.6.2
- 你使用的**Tensorflow**版本: 1.12.0
- 你使用的**Keras**版本: 2.2.4
- 你使用的**bert4keras**版本:0.5.8-0.8.1 
- 你使用纯**keras**还是**tf.keras**: keras
- 你加载的**预训练模型**:bert

### 核心代码
```python
# 请在此处贴上你的核心代码。
# 请尽量只保留关键部分，不要无脑贴全部代码。
```
### 输出信息
```shell
 [[node replica_0/model_2/model_1/Embedding-Token/strided_slice (defined at /home/chubingbing/soft/anaconda3/envs/py3.6/lib/python3.6/site-packages/bert4keras/layers.py:88)  = StridedSlice[Index=DT_INT32, T=DT_BOOL, begin_mask=3, ellipsis_mask=0, end_mask=1, new_axis_mask=0, shrink_axis_mask=0, _device="/device:GPU:0"](replica_0/model_2/model_1/Embedding-Token/NotEqual, replica_0/model_2/model_1/Embedding-Token/strided_slice/stack, replica_0/model_2/model_1/Embedding-Token/strided_slice/stack_1, replica_0/model_2/model_1/Embedding-Token/strided_slice/stack_2)]]
```
![image](https://user-images.githubusercontent.com/13550295/85359992-7d24ea00-b54a-11ea-9226-5795c58af6ba.png)

### 自我尝试
不管什么问题，请先尝试自行解决，“万般努力”之下仍然无法解决再来提问。此处请贴上你的努力过程。

1.  使用0.5.8可以支持多gpu训练、预测、保存等都没有问题，但是使用0.8.1不可以，然后尝试了0.6.1，发现也不支持，推测应该是**自从0.6.1不再支持多gpu**。
2. 触发的代码是 layers.py的88行，Embedding里面compute_mask方法，mask1 = K.ones_like(mask[:, :1], dtype='bool')，

删除线内容需要待确认：
~~应该是 **K.ones_like**不支持多gpu，因为我在做分类任务，尝试使用以下代码，出现错误，但是如果把K.zeros_like(x)，换成代码片段2，在0.5.8是可以正常多gpu运行的~~

~~# 代码片段1，0.5.8版本多gpu出错：~~
~~input = Input(shape=(None,))~~
~~transformer_model = build_transformer_model(config_path, checkpoint_path)~~
~~# x为bert分词后的token id，K.zeros_like(x)为segment id~~
~~encoder = transformer_model(Lambda(lambda x: [x, K.zeros_like(x)])(input))~~
~~encoder = Lambda(lambda x: x[:, 0])(encoder)~~

```
# 代码片段2：0.5.8版本多gpu正常
first = Input(shape=(None,))
second = Input(shape=(None,))
input = [first, second]
encoder = transformer_model(input)
encoder = Lambda(lambda x: x[:, 0])(encoder)
```

**Comment by bojone** [comment]
你的多gpu是怎么做的？是keras自带的`multi_gpu_model`方法？

**Comment by tiandiweizun** [comment]
是的，我更新了issue，以前描述有错误
    ```
    model.summary()
    new_model = model
    if (gpu_size > 1):
        new_model = multi_gpu_model(model, gpus=gpu_size)  # 设置使用多个gpu，该句放在模型compile之前
    new_model.compile(loss='sparse_categorical_crossentropy', optimizer=optimizer, metrics=['accuracy'])
```

**Comment by bojone** [comment]
说实话我用`multi_gpu_model`就没成功过。我建议用tf.keras的`MirroredStrategy`，tf 1.14后面的版本都可以用。参考：https://tf.wiki/zh_hans/appendix/distributed.html

**Comment by tiandiweizun** [comment]
好的
